File: DotNetGotchas/CSharp/HandleException/UnhandledExceptions/ALibrary/Utility.cs
//Utility.cs part of ALibrary.dll
using System;

namespace ALibrary
{
	public class Utility
	{
		public double Operate(int value1, int value2)
		{
			// Some operation
			// Of course, this is an enormous programming error
			// Never do division without making sure the denominator
			// is not zero.
			// We’re just doing it here for the sake of example.
			double val = value1 / value2;
			return Math.Sqrt(val);
		}
	}
}

File: DotNetGotchas/VBDN/HandleException/UnhandledExceptions/ALibrary/Utility.vb
'Utility.vb part of ALibrary.dll

Public Class Utility
    Public Function Operate( _
        ByVal value1 As Integer, ByVal value2 As Integer) As Double
        'Some operation
        ' Of course, this is an enormous programming error
        ' Never do division without making sure the denominator
        ' is not zero.
        ' We’re just doing it here for the sake of example.
        Dim val As Double = value1 / value2

        If Double.IsInfinity(val) Then
            Throw New DivideByZeroException( _
                "Attempted to Divide by Zero")
        End If
        Return Math.Sqrt(val)
    End Function
End Class
